File: src/main/java/de/rubixdev/enchantedshulkers/enchantment/AugmentEnchantment.java
package de.rubixdev.enchantedshulkers.enchantment;

import de.rubixdev.enchantedshulkers.Mod;
import de.rubixdev.enchantedshulkers.config.WorldConfig;

public class AugmentEnchantment extends ContainerEnchantment {
    public AugmentEnchantment() {
        super(Mod.AUGMENTABLE_CONTAINER_TARGET);
    }

    @Override
    protected boolean generate() {
        return WorldConfig.generateAugment();
    }

    @Override
    public int getMaxLevel() {
        return WorldConfig.maxAugmentLevel();
    }
}

File: src/main/kotlin/de/rubixdev/enchantedshulkers/enchantment/AugmentEnchantment.kt
package de.rubixdev.enchantedshulkers.enchantment

import de.rubixdev.enchantedshulkers.Mod
import de.rubixdev.enchantedshulkers.config.WorldConfig

class AugmentEnchantment : ContainerEnchantment(
    //#if MC >= 12006
    Mod.AUGMENTABLE_CONTAINER_TAG,
    //#else
    //$$ Mod.AUGMENTABLE_CONTAINER_TARGET,
    //#endif
    //#if MC >= 12006
    WorldConfig.maxAugmentLevel,
    //#endif
) {
    override fun generate() = WorldConfig.generateAugment
    override fun getMaxLevel() = WorldConfig.maxAugmentLevel
}
